hack/template-lint.sh

```diff
@@ -25,6 +25,16 @@ do
         continue
     fi
 
+    # Handle some special case files that live in the enhancements
+    # directory but are not enhancements, and cannot be moved
+    # elsewhere.
+    case "${file}" in
+        enhancements/operator-dev-doc.md | enhancements/README.md | enhancements/rebase.md)
```

Comment:
`enhancements/README.md` is old, and folks linking should have had time to update by now.  Can we just remove it?.

The other two seem like they could move into the root for like `CONVENTIONS.md` or go in a non-enhancement dir (`docs/`?).  I don't feel too strongly about it, so feel free to ignore and land as this stands, but personally I'd rather move non-enhancement elsewhere instead of leaving them under `enhancements/` and then enumerating them here.

Reply:
Yeah, I wanted to move them, too. @atiratree pointed out several places that link to operator-dev-doc and I didn't want to hold up #794 while we figured out what to do about all of those other references. I definitely don't want to add anything else to this case statement. 

Let me take a few minutes to find the references. If there are only a few, I can move the file and submit patches to fix the links.

Reply:
OK, I didn't find very many references so next week I'll fix them.

```
$ for d in *; do git -C $d grep operator-dev-doc.md && (echo $d; echo); done
README.md:[OpenShift ClusterOperator](https://github.com/openshift/enhancements/blob/master/enhancements/operator-dev-doc.md#what-is-an-openshift-clusteroperator).
README.md:[OpenShift ClusterOperator and Operand Developer Document](https://github.com/openshift/enhancements/blob/master/enhancements/operator-dev-doc.md#how-do-i-buildupdateverifyrun-unit-tests)
README.md:[Testing a ClusterOperator/Operand image in a cluster](https://github.com/openshift/enhancements/blob/master/enhancements/operator-dev-doc.md#how-can-i-test-changes-to-an-openshift-operatoroperandrelease-component)
cluster-authentication-operator

README.md:[OpenShift ClusterOperator](https://github.com/openshift/enhancements/blob/master/enhancements/operator-dev-doc.md#what-is-an-openshift-clusteroperator).
cluster-csi-snapshot-controller-operator

README.md:[OpenShift ClusterOperator](https://github.com/openshift/enhancements/blob/master/enhancements/operator-dev-doc.md#what-is-an-openshift-clusteroperator).
README.md:[OpenShift ClusterOperator and Operand Developer Document](https://github.com/openshift/enhancements/blob/master/enhancements/operator-dev-doc.md#how-do-i-buildupdateverifyrun-unit-tests)
README.md:[Testing a ClusterOperator/Operand image in a cluster](https://github.com/openshift/enhancements/blob/master/enhancements/operator-dev-doc.md#how-can-i-test-changes-to-an-openshift-operatoroperandrelease-component)
cluster-openshift-controller-manager-operator

README.md:[OpenShift ClusterOperator](https://github.com/openshift/enhancements/blob/master/enhancements/operator-dev-doc.md#what-is-an-openshift-clusteroperator).
README.md:[OpenShift ClusterOperator and Operand Developer Document](https://github.com/openshift/enhancements/blob/master/enhancements/operator-dev-doc.md#how-do-i-buildupdateverifyrun-unit-tests)
README.md:[Testing a ClusterOperator/Operand image in a cluster](https://github.com/openshift/enhancements/blob/master/enhancements/operator-dev-doc.md#how-can-i-test-changes-to-an-openshift-operatoroperandrelease-component)
cluster-policy-controller

README.md:[OpenShift ClusterOperator](https://github.com/openshift/enhancements/blob/master/enhancements/operator-dev-doc.md#what-is-an-openshift-clusteroperator)
README.md:[OpenShift ClusterOperator and Operand Developer Document](https://github.com/openshift/enhancements/blob/master/enhancements/operator-dev-doc.md#how-do-i-buildupdateverifyrun-unit-tests)
README.md:[Testing a ClusterOperator/Operand image in a cluster](https://github.com/openshift/enhancements/blob/master/enhancements/operator-dev-doc.md#how-can-i-test-changes-to-an-openshift-operatoroperandrelease-component)
service-ca-operator
```

Reply:
I also usually pin my GitHub links to a particular commit message, instead of floating with `master`, which would have protected these from the move, subsequent renames, etc.  Of course, then folks have to find their way from the linked content to the current analog on their own, but at least they have the frozen content to start with.